base/src/main/java/com/thoughtworks/go/util/SystemEnvironment.java

```diff
@@ -181,6 +181,8 @@
     public static GoSystemProperty<String> GO_UPDATE_SERVER_URL = new GoStringSystemProperty("go.update.server.url", "https://update.gocd.org/channels/supported/latest.json");
     public static GoSystemProperty<Boolean> GO_CHECK_UPDATES = new GoBooleanSystemProperty("go.check.updates", true);
 
+    public static GoSystemProperty<String> GO_DATA_SHARING_SERVER_URL = new GoStringSystemProperty("go.datasharing.server.url", "https://datasharing.gocd.org/v1");
```

Comment:
The Data Sharing Server URL will change.

@arvindsv @jyotisingh -- we can decide later on what should be the data sharing URL.

Reply:
There is no reason for `GO_DATA_SHARING_SERVER_URL` being a configurable system property.
Can it be a final string defined in `SystemEnvironment`?

If any user defines `go.datasharing.server.url` with a new value, the GoCD data collecting server will stop receiving data even if the data sharing consent is set to true.

@arvindsv @jyotisingh -- thoughts?

Reply:
Ofcourse, for now this value for url seems fine. 

I think its fine to leave it as a `SystemEnvironment` property. The only reason for a user to redefine `go.datasharing.server.url` would be to stop sending reports to our server, so in that sense, for us it's as good as them not giving a consent and there's nothing we can do about it. Leaving it as a `SystemEnvironment` gives us some flexibility. At a later point, if we think we do not need that flexibility, we can get rid of the property and make it a constant instead.